deepchecks/nlp/checks/data_integrity/conflicting_labels.py

```diff
@@ -82,30 +82,69 @@ def _text_normalization_kwargs(self):
     def _truncate_text(self, x: str) -> str:
         return truncate_string(x, self.max_text_length_for_display)
 
+    def _get_labels(self, dataset):
+        if dataset.task_type is TaskType.TOKEN_CLASSIFICATION:
+            labels = [tuple(t.cast(t.Sequence[t.Any], it)) for it in dataset.label]
+        elif dataset.is_multi_label_classification():
+            labels = [tuple(np.where(row == 1)[0]) for row in dataset.label]
+        elif dataset.task_type is TaskType.TEXT_CLASSIFICATION:
+            labels = dataset.label
+        else:
+            raise DeepchecksValueError(f'Unknown task type - {dataset.task_type}')
+        return labels
+
+    @staticmethod
+    def _get_conflicting_indices(df: pd.DataFrame):
+        by_hash = df.loc[:, ['hash', 'Label']].groupby(['hash'], dropna=False)
+        count_labels = lambda x: len(set(x.to_list()))
+        n_of_labels_per_sample = by_hash['Label'].aggregate(count_labels)
+
+        ambiguous_samples_hashes = n_of_labels_per_sample[n_of_labels_per_sample > 1]
+        return frozenset(ambiguous_samples_hashes.index.to_list())
+
     def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         """Run check."""
         dataset = context.get_data_by_kind(dataset_kind)
         dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
         dataset = t.cast(TextData, dataset)
-        samples = dataset.text
-        n_of_samples = len(samples)
+        n_of_samples = len(dataset)
 
         if n_of_samples == 0:
             raise DeepchecksValueError('Dataset cannot be empty')
 
+        # Reduce dataset by first checking on truncated strings
+        truncated_samples = [cut_string(x) for x in dataset.text]
+        truncated_samples_hashes = hash_samples(normalize_samples(
+            truncated_samples,
+            **self._text_normalization_kwargs
+        ))
+
+        labels = self._get_labels(dataset)
+
+        df = pd.DataFrame({
+            'hash': truncated_samples_hashes,
+            'Sample ID': dataset.get_original_text_indexes(),
+            'Label': labels,
+            'Text': dataset.text,
+        })
```

Comment:
Why not take this out into an outer function and just run it twice? seems like a lot of code duplication

Reply:
I mean, I know you did that. But even the hashing and creation of the df